palamedes/align.py

```diff
@@ -106,11 +106,30 @@ def merge_variant_blocks(left: VariantBlock, right: VariantBlock) -> VariantBloc
     return VariantBlock(new_alignment_block, new_reference_blocks, new_alternate_blocks)
 
 
-def merge_reduce(blocks: list[VariantBlock], next_block: VariantBlock) -> list[VariantBlock]:
-    """Helper function to implement merge or append, passed to functools.reduce to merge blocks"""
+def merge_reduce(
+    blocks: list[VariantBlock], next_block: VariantBlock, split_consecutive_mismatches: bool = False
+) -> list[VariantBlock]:
+    """
+    Helper function to implement merge or append, passed to functools.reduce to merge blocks. Optional
+    split_consecutive_mismatches will keep consecutive mismatches from being merged, allowing for a
+    chain of mismatches vs a delins. Note that this is non-standard behavior according to HGVS.
+    """
     peek_block = blocks[-1]
 
-    if can_merge_variant_blocks(peek_block, next_block):
+    # base check for merge-ability
+    base_can_merge = can_merge_variant_blocks(peek_block, next_block)
+
+    # extra check for flag, merge if false OR merge if true and left and right
+    # are not single bp mismatches. If true and both are single base mismatches
+    # then we do not merge, this 'splitting'
+    mismatch_split_can_merge = not split_consecutive_mismatches or all(
+        [
+            peek_block.alignment_block.bases != VARIANT_BASE_MISMATCH,
+            next_block.alignment_block.bases != VARIANT_BASE_MISMATCH,
+        ]
+    )
```

Comment:
Logic is a bit gnarly here, but basically you check if they could be merged using the default logic, and then if the flag is passed, you see if they are both single mismatches. If they are, then you dont merge.